src/CommunityToolkit.Maui/Views/RatingView/RatingView.Shared.cs

```diff
@@ -0,0 +1,361 @@
+	public ICommand? Command
+	{
+		get => (ICommand)GetValue(CommandProperty);
+		set => SetValue(CommandProperty, value);
+	}
+
+	///<summary></summary>
+	public object? CommandParameter
+	{
+		get => GetValue(CommandParameterProperty);
+		set => SetValue(CommandProperty, value);
+	}
+
+	///<summary></summary>
+	public RatingShape Shape
+	{
+		get => (RatingShape)GetValue(ShapeProperty);
+		set => SetValue(ShapeProperty, value);
+	}
+
+	///<summary></summary>
+	public new View? Content { get => base.Content; internal set => base.Content = value; }
+
+	///<summary></summary>
+	public RatingView()
+	{
+		shapes = new Microsoft.Maui.Controls.Shapes.Path[MaximumRating];
+
+
+		HorizontalOptions = LayoutOptions.CenterAndExpand;
+
+		this.innerContent.ColumnSpacing = Spacing;
+
+		InitializeShape();
+
+		DrawBase();
+
+
+		this.Stroke = Colors.Transparent;
+	}
+
+	static void OnBindablePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+	{
+		//re-draw forms.
+		((RatingView)bindable).ReDraw();
+	}
+
+	static void OnShapePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+	{
+		((RatingView)bindable).InitializeShape();
+		((RatingView)bindable).ReDraw();
+	}
+
+	void OnShapeTapped(object? sender, TappedEventArgs? e)
+	{
+		var tappedShape = sender as Microsoft.Maui.Controls.Shapes.Path;
+
+
+		if (tappedShape is null) { return; }
+
+		var columnIndex = innerContent.GetColumn(tappedShape);
+
+
+		if (MaximumRating > 1)
+		{
+			CurrentRating = columnIndex + 1;
+		}
+
+		var data = new Rating
+		{
+			CurrentRating = CurrentRating,
+			CommandParameter = CommandParameter
+		};
+
+		if (Command is not null && Command.CanExecute(data))
+		{
+			Command.Execute(data);
+		}
+	}
+
+	void DrawBase()
+	{
+		for (int i = 0; i < MaximumRating; i++)
+		{
+			innerContent?.ColumnDefinitions.Add(new ColumnDefinition { Width = Size });
+
+			Microsoft.Maui.Controls.Shapes.Path image = new();
+			image.Data = (Geometry?)new PathGeometryConverter().ConvertFromInvariantString(shape);
+			
+			if (i <= CurrentRating)
+			{
+
+				image.Fill = FilledBackgroundColor;
+				image.Stroke = FilledBackgroundColor;
+				image.Aspect = Stretch.Uniform;
+				image.HeightRequest = Size;
+				image.WidthRequest = Size;
+			}
+			else
+			{
+				image.Fill = EmptyBackgroundColor;
+				image.Stroke = StrokeColor;
+
+				image.StrokeLineJoin = PenLineJoin.Round;
+				image.StrokeLineCap = PenLineCap.Round;
+				image.StrokeThickness = StrokeThickness;
+
+				image.Aspect = Stretch.Uniform;
+				image.HeightRequest = Size;
+				image.WidthRequest = Size;
+			}
+
+
+			if (ShouldAllowRating)
+			{
+				var tapGestureRecognizer = new TapGestureRecognizer();
+				tapGestureRecognizer.Tapped += OnShapeTapped;
+				image.GestureRecognizers.Add(tapGestureRecognizer);
+			}
+
+			innerContent?.Children.Add(image);
+			innerContent?.SetColumn(image, i);
+
+			shapes[i] = image;
+		}
+
+
+		UpdateDraw();
+	}
+
+	void ReDraw()
```

Comment:
DrawBase looks almost the same as this method. Consider extracting the common part to a separate method